M2/Macaulay2/packages/Macaulay2Doc/functions/intersect-doc.m2

```diff
@@ -2,28 +2,52 @@
 --- author(s):Giulio
 --- notes: 
 
-document { 
-     Key => {intersect,(intersect, List),(intersect,Sequence)},
-     Headline => "compute an intersection",
-     Usage => "intersect (M,N,...,P) ",
-     Inputs => { Nothing => { TT "(M,N,...,P)", ", ", ofClass {List, Sequence}, " of modules or ideals that are submodules of the same module
-	       or ideals in the same ring" } },
-     Outputs => { {ofClass Module, " or ", ofClass Ideal," that is the intersection of the elements in the list or in the sequence."} },
-     "This function calculates the intersection of submodules of the same free module, or of ideals in the same ring.",
-     PARA "The following example computes the intersection of a sequence of ideals.",
-     EXAMPLE {
-	  "R=ZZ/101[a..d];",
-	  "I=intersect(ideal(a,b),ideal(b,c),ideal(c,d),ideal(d,a))"
-	  },
-     PARA "The following example computes the intersection of a list of modules.",
-     EXAMPLE {
-	  "R=ZZ[x,y,z];",
-	  "M=image matrix{{3*x},{3*x}};", 
-	  "N=image matrix{{5*y},{5*y}};",
-	  "P=image matrix{{7*z},{7*z}};",
-	  "intersect{M,N,P}"
-	  },
-     PARA {"The command ", TO "intersect", " will only work with proper
-     ideals. To intersect an ideal with a ring, use ",
-     TO "selectInSubring", " along with the elimination ordering, see ", TO "Eliminate", "."}
-     }
+
+doc ///
+Node
+  Key
+    intersect
+  Headline
+    compute an intersection
+Node
+  Key
+    (intersect, List)
+    (intersect, Sequence)
+    (intersect, Ideal)
+    (intersect, Module)
+    [intersect, Strategy]
+    [intersect, MinimalGenerators]
+  Headline
+    compute an intersection of ideals or modules
+  Usage
+    intersect(M, N, ..., P)
+  Inputs
+    :{List,Sequence}
+      containing modules that are submodules of the same module or ideals in the same ring
+    Strategy=>Thing
+      specifies the algorithm
+    MinimalGenerators=>Boolean
+      indicates whether the output should be @TO2 {trim, "trimmed"}@
+  Outputs
+    :{Ideal,Module}
+      the intersection of the objects given
+  Description
+    Text
+      This function calculates the intersection of submodules of the same free module, or of ideals in the same ring.
+
+      The following example computes the intersection of a sequence of ideals.
+    Example
+      R=ZZ/101[a..d];
+      I=intersect(ideal(a, b), ideal(b, c), ideal(c, d), ideal(d, a))
+    Text
+      The following example computes the intersection of a list of modules.
+    Example
+      R=ZZ[x, y, z];
+      M=image matrix{{3*x}, {3*x}};
+      N=image matrix{{5*y}, {5*y}};
+      P=image matrix{{7*z}, {7*z}};
+      intersect{M, N, P}
+    Text
+      The command @TO "intersect"@ will only work with proper ideals. To intersect an ideal with a ring,
```

Comment:
Actually, "intersect" works fine even if some of the ideals are the unit ideal.

Reply:
The point is that you can't intersect with a subring. This sentence exists in [documentation of v1.16](http://www2.macaulay2.com/Macaulay2/doc/Macaulay2-1.16/share/doc/Macaulay2/Macaulay2Doc/html/_intersect.html) as well, and I mainly changed it to  the SimpleDoc format, so I'm hesitant to change it myself. But if you have an alternative phrasing in mind, feel free to suggest it or change it.